bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -55,9 +60,53 @@ public static String getLocalIpv4HostAddress() {
             }
             return hostAddress;
         } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: " + ex.getMessage(), ex);
+            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
             return null;
         }
     }
 
+    /**
+     * Get all broadcast addresses on the current host
+     *
+     * @return list of broadcast addresses
+     */
+    public static List<String> getAllBroadcastAddresses() {
+        try {
+            List<String> broadcastAddresses = new ArrayList<String>();
```

Comment:
Isn't a `LinkedList` better if we don't know the size (to reduce reallocations and copies)?